src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -640,7 +640,7 @@ private int decodePublish(
                     }
                 }
 
-                assert publisher != null;
+                publisher.updatePublishCapabilityIfNecessary(traceId, authorization);
```

Comment:
What happens if the subscribe stream is still active when the publisher times out?
Please check how the flow control behaves to make sure we are not sending `DATA` after a `FLUSH` that switched off `publish` capability without first receiving `WINDOW` after switching back on `publish` capability`.